src/models/Users.js

```diff
@@ -317,4 +317,30 @@ export default class Users extends Model {
     const secret = `${this.password}!${this.createdAt.toISOString()}`;
     return jwt.verify(token, secret);
   }
+
+  /**
+   * method to upsert subscription
+   *
+   * @static
+   * @memberof Users
+   *
+   * @param {Array} categoryIds Array of category Ids to be subscribed to
+   *
+   * @returns {Object} Subscription data
+   */
+  async bulkUpsertSubscriptions(categoryIds) {
+    let data;
+    data = categoryIds.map(async (categoryId) => {
```

Comment:
Fun fact, you can also use reduce for this. You knowhow reduce has to wait for the previous execution to be completed before continuing, you can leverage that behaviour. Not saying you should change this though, it's just another option